### DISCUSSION TITLE
Variable - Clipboard contents

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1126] [Suggestions]
Suggesting to add a variable for text based clipboard content.
This would provide text based data saved in clipboard to be able to be used in internal DC based commands.

Single row text would just be added as a variable.

If clipboard contained multiple rows of text would have to decide how this would be handled....
-Ignore if multi row text
-Show warning message letting user know that clipboard contains more than one row of text
-Just use first row of text

Thank You

**Comment by advice2020** [comment]
Surprised no one ever agreed with this one?
At the time when I mentioned this I was not fully aware of how Linux had some strange limitations with how it stores clipboard text content, does anyone know if this is why this variable is not included?

Thank you

**Reply by j2969719** [reply]
its simply easier to check the contents or make the necessary string transformations with a luascript https://doublecmd.github.io/doc/en/lua.html#libraryclip than with hardcoded variable, since a call can occur when the clipboard may or may not contain what is needed

**Comment by advice2020** [comment]
@j2969719 
Yeah it is "simply easier" for you because you know what you are doing but this requested variable would be for the rest us who do not :)
Well actually I am the only one interested in this on this site, so replace "the rest of us who do not" with "me who does not" :)
Thank You

**Comment by Skif-off** [comment]
`%[message;default_val]` from [Variables in parameters > 13. Interaction](https://doublecmd.github.io/doc/en/variables.html#interaction) + Ctrl+V?

**Reply by advice2020** [reply]
Thank you for your response.
Temporary solution but feel it could be smoother if value was just applied.  But what you mention does touch upon what j2969719 mentions about being able to see the text value (if other users require this).
This would allow functionality similar to CL commands such as "xclip" but would be internal in DC without the need for 3rd party commands.
Anyways, thank you for mentioning a solution.
